src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -4329,6 +4329,584 @@ static void Main()
             Assert.Equal(FlowAnalysisAnnotations.NotNullWhenTrue, lambda.Parameters[0].FlowAnalysisAnnotations);
         }
 
+        [Fact]
+        public void LambdaReturnType_01()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void F<T>()
+    {
+        Func<T> f1 = T () => default;
+        Func<T, T> f2 = T (x) => { return x; };
+        Func<T, T> f3 = T (T x) => x;
+    }
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (6,22): error CS8652: The feature 'lambda return type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         Func<T> f1 = T () => default;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "T").WithArguments("lambda return type").WithLocation(6, 22),
+                // (7,25): error CS8652: The feature 'lambda return type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         Func<T, T> f2 = T (x) => { return x; };
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "T").WithArguments("lambda return type").WithLocation(7, 25),
+                // (8,25): error CS8652: The feature 'lambda return type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         Func<T, T> f3 = T (T x) => x;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "T").WithArguments("lambda return type").WithLocation(8, 25));
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void LambdaReturnType_02()
```

Comment:
Could you add a similar test, but with one of the type parameters constrainted by the other type parameter (`where T : U` for example)?